### PR TITLE
Add glance metric component

### PR DESCRIPTION
**Pull request by maxgds**
This is the glance metric component for displaying metrics at the top of our single content item page. 

https://trello.com/c/xWh1yAwa/603-3-build-glance-metrics-component
screen shot 2018-08-22 at 11 27 25

### CODE REVIEW

File: app/assets/stylesheets/components/_glance-metric.scss [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  @include media-down(tablet) {
+    min-height: 2.8em;
+  }
+
+  @include media(mobile) {
+    min-height: 0;
+  }
+}
+
+.app-c-glance-metric__trend__text {
```
Comment (alex-ju) [inline app/assets/stylesheets/components/_glance-metric.scss]: According to BEM this should be `__trend-text` as it doesn't allow chaining elements.

File: app/assets/stylesheets/components/_glance-metric.scss [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  @include govuk-visually-hidden;
+}
+
+.app-c-glance-metric__trend__icon {
```
Comment (alex-ju) [inline app/assets/stylesheets/components/_glance-metric.scss]: `__trend-icon`

File: app/views/components/_glance-metric.html.erb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  link_url ||= false
+  measurement_display_label ||= ""
+  measurement_explicit_label ||= ""
+  context ||= ""
+  trend_percentage ||= ""
+%>
+<% if name && figure && period %>
+  <div class="app-c-glance-metric govuk-body">
+    <h3 class="app-c-glance-metric__heading govuk-heading-s"><%= name %></h3>
+    <span class="app-c-glance-metric__figure govuk-!-font-weight-bold govuk-!-font-size-36 "><%= figure %><span class="app-c-glance-metric__measurement" <% if measurement_explicit_label %>aria-label="<%= measurement_explicit_label %>"<% end %>><%= measurement_display_label %></span></span>
```
Comment (alex-ju) [inline app/views/components/_glance-metric.html.erb]: I would avoid using overwrites here.
We can define the font styling in `__figure` using a mixin: `govuk-font($size: 36, $weight: bold)`

File: spec/rails_helper.rb~Chart component [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# This file is copied to spec/ when you run 'rails generate rspec:install'
```
Comment (alex-ju) [inline spec/rails_helper.rb~Chart component]: We don't want this one in, do we?

**Comment by maxgds** [comment]
Thanks @alex-ju - useful pointers, changes made. Could you re-check and approve if happy?

**Pull request merged**